### PR TITLE
fix issue 3888

### PR DESCRIPTION
**Pull request by drafish**
fix issue #3888

### CODE REVIEW

**Comment by fwx5618177** [comment]
Looks better. @joeizang PTAL?

File: libs/remix-ui/run-tab/src/lib/actions/deploy.ts [hunk 1]
```diff
@@ -155,14 +155,14 @@ export const createInstance = async (
...
-  const finalCb = (error, contractObject, address) => {
+  const finalCb = async (error, contractObject, address) => {
     if (error) {
       const log = logBuilder(error)
 
       return terminalLogger(plugin, log)
     }
     addInstance(dispatch, { contractData: contractObject, address, name: contractObject.name })
-    const data = plugin.compilersArtefacts.getCompilerAbstract(contractObject.contract.file)
+    const data = await plugin.compilersArtefacts.getCompilerAbstract(contractObject.contract.file)
```
Comment (yann300) [inline libs/remix-ui/run-tab/src/lib/actions/deploy.ts:165]: Thanks for this fix! could you also check if  `getCompilerAbstract` is missing the `await` in other places in the code?

Comment (drafish) [reply libs/remix-ui/run-tab/src/lib/actions/deploy.ts:165]: OK, I will check it.

File: apps/remix-ide/src/app/udapp/run-tab.js [hunk 2]
```diff
@@ -182,8 +182,8 @@ export class RunTab extends ViewPlugin {
     return this.call('fileManager', 'readFile', fileName)
   }
 
-  resolveContractAndAddInstance (contractObject, address) {
-    const data = this.compilersArtefacts.getCompilerAbstract(contractObject.contract.file)
+  async resolveContractAndAddInstance (contractObject, address) {
+    const data = await this.compilersArtefacts.getCompilerAbstract(contractObject.contract.file)
```
Comment (drafish) [inline apps/remix-ide/src/app/udapp/run-tab.js:186]: Here is another place that is missing await. Nowhere else. @yann300

File: apps/remix-ide/src/app/udapp/run-tab.js
Comment (drafish) [inline apps/remix-ide/src/app/udapp/run-tab.js:1]: Here is another place that is missing await. Nowhere else. @yann300

File: apps/remix-ide/src/app/udapp/run-tab.js [hunk 3]
```diff
@@ -182,8 +182,8 @@ export class RunTab extends ViewPlugin {
...
 
     this.compilersArtefacts.addResolvedContract(addressToString(address), data)
```
Comment (drafish) [inline apps/remix-ide/src/app/udapp/run-tab.js:188]: Here is another place that is missing await. Nowhere else. @yann300

**Review by yann300 — APPROVED** [review]

**Comment by Aniket-Engg** [comment]
@drafish can you rebase it?

**Comment by drafish** [comment]
rebased @Aniket-Engg

**Comment by Aniket-Engg** [comment]
Looks like you need to rebase again @drafish

**Comment by drafish** [comment]
rebased again @Aniket-Engg

**Comment by Aniket-Engg** [comment]
Thanks @drafish !!!

**Pull request merged**